### PR TITLE
docs: update roadmap of 2024

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-docs

#### What this PR does / why we need it:
Update roadmap of 2024.

#### Special notes for your reviewer:
The English version is translated by Copilot, please help check it @Jane201510

### CODE REVIEW

**Review by yizhouxw — APPROVED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -148,15 +148,14 @@ For detailed guidelines on contributing, please refer to the [Contribution Guide
...
+| **Ease of Use**                    | <ul><li> Optimize SQL development page interaction</li><li> Optimize database search capability</li><li> Desktop version supports custom JVM parameters</li><li> Support custom configuration items (including shortcuts, etc.)</li></ul>                     | <ul><li> UI theme renewal</li><li> New project control concept guidance</li><li> Optimize work order management page interaction</li><li> Strengthen code automatic completion capability</li></ul>                                                                                             | <ul><li> Optimize PL development experience</li><li> Optimize wide table scenario page interaction</li><li> Desktop version startup speed optimization to within 20s</li></ul>                                                                                                        | <ul><li> Provide template SQL formatting capability</li><li> Optimize operation record page interaction</li></ul>                                                                                              |
```
Comment (JessieWuJiexi) [inline README.md:158]: 1. Optimize work order management page interaction -> Optimize ticket management page interaction
2. Strengthen code automatic completion capability -> **Enhance** code auto-completion capability
3. Desktop version startup speed optimization to within 20s -> Desktop version startup speed **optimized** to within 20s
4. Provide template SQL formatting capability -> Provide SQL formatting capability based on templates

File: README-zh-CN.md [hunk 2]
```diff
@@ -142,15 +142,14 @@ ODC 期望构建一个开放的社区，一起打造数据库开发管控协同
...
+| **变更更风险管控**   | <ul><li> 支持库级权限访问控制</li><li> 支持任务异常发送告警</li><li> 支持自动识别 offline DDL 变更</li></ul>                                 | <ul><li> 支持逻辑库 DDL 变更；</li><li> 支持无锁数据变更</li><li> 支持多库（pipeline）变更</li><li> 支持配置库管理员</li><li> 支持表级权限访问控制</li><li> 新增全局项目管理员</li></ul> | <ul><li> 支持基于状态的结构变更</li><li> 支持逻辑库 DML/DQL</li><li> 支持模版化配置 SQL-check</li><li> 支持敏感列权限申请</li><li> 提供基于行级回收站的变更回滚能力</li></ul> | <ul><li> 支持自动识别敏感数据</li><li> 支持基于状态的结构变更</li><li> 命令行窗口行为纳入操作审计</li></ul>                    |
+| **数据生命周期管理**  | <ul><li> 数据归档/清理支持 SQL 预览</li><li> 数据归档/清理支持唯一索引表</li><li> 数据归档支持源表目标表同库</li><li> 分区计划支持 OB Oracle 数据源</li></ul> | <ul><li> 数据归档源端查询性能优化</li><li> 支持从 OB 归档到文件/oss/s3</li><li> 同构数据归档支持 DDL 增量变更同步</li></ul>                                             | <ul><li> 支持从 OB 归档到文件COS/OBS</li><li> 支持异构数据库归档表结构同步</li><li> 支持关联（外键）表场景进行归档</li><li> 数据归档/清理风险提示</li></ul>                  | <ul><li> 据归档/清理支持逻辑库</li><li> 据归档/清理支持 Oracle 数据源</li></ul>                                  |
```
Comment (JessieWuJiexi) [inline README-zh-CN.md:151]: 1. 文件COS/OBS -> 文件/COS/OBS
2. 最后两条都少了一个“数”字：<li> 据归档/清理支持逻辑库</li><li> 据归档/清理支持 Oracle 数据源</li>

Comment (smallsheeeep) [reply README-zh-CN.md:151]: Okay, I modified it.

File: README.md [hunk 3]
```diff
@@ -148,15 +148,14 @@ For detailed guidelines on contributing, please refer to the [Contribution Guide
...
+| **HTAP Collaborative Development** | <ul><li> Support Doris data source</li><li> Support Oracle data source</li><li> Support homogeneous database structure comparison/synchronization</li><li> Built-in common code snippets</li></ul>                                                            | <ul><li> Support natural language conversion to SQL</li><li> Support real-time query analysis</li><li> Support result set conversion to chart</li><li> Support global database object retrieval and positioning</li><li> Provide dbt-oceanbase plug-in</li></ul>                                | <ul><li> Support Git repository integration</li><li> Support dbt development integration</li><li> Support natural language conversion to chart</li><li> Support project internal script and code snippet sharing</li></ul>                                                            | <ul><li> Support AI task scheduling</li><li> Support DAG task scheduling</li><li> Support lightweight report management</li><li> Support heterogeneous database structure comparison/synchronization</li></ul> |
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply README.md:155]: Thx! I modified it.

File: README.md [hunk 4]
```diff
@@ -148,15 +148,14 @@ For detailed guidelines on contributing, please refer to the [Contribution Guide
...
+| **Change and Risk Control**        | <ul><li> Support database-level access control</li><li> Support task exception sending alarm</li><li> Support automatic identification of offline DDL changes</li></ul>                                                                                       | <ul><li> Support logical database DDL changes;</li><li> Support lock-free data changes</li><li> Support multi-database (pipeline) changes</li><li> Support configuration database administrator</li><li> Support table-level access control</li><li> Add global project administrator</li></ul> | <ul><li> Support state-based structural changes</li><li> Support logical database DML/DQL</li><li> Support template configuration SQL-check</li><li> Support sensitive column access application</li><li> Provide change rollback capability based on row-level recycle bin</li></ul> | <ul><li> Support automatic identification of sensitive data</li><li> Support state-based structural changes</li><li> Command window behavior is included in operation audit</li></ul>                          |
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply README.md:156]: Modified.

File: README.md [hunk 5]
```diff
@@ -148,15 +148,14 @@ For detailed guidelines on contributing, please refer to the [Contribution Guide
...
+| **Data Lifecycle Management**      | <ul><li> Data archiving/cleaning supports SQL preview</li><li> Data archiving/cleaning supports unique index tables</li><li> Data archiving supports source table target table same database</li><li> Partition plan supports OB Oracle data source</li></ul> | <ul><li> Data archiving source query performance optimization</li><li> Support archiving from OB to file/oss/s3</li><li> Homogeneous data archiving supports DDL incremental change synchronization</li></ul>                                                                                   | <ul><li> Support archiving from OB to file COS/OBS</li><li> Support heterogeneous database archiving table structure synchronization</li><li> Support association (foreign key) table scenario archiving</li><li> Data archiving/cleaning risk prompts</li></ul>                      | <ul><li> Data archiving/cleaning supports logical database</li><li> Data archiving/cleaning supports Oracle data source</li></ul>                                                                              |
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply README.md:157]: Modified.

File: README.md (hunk 1 above)
Comment (smallsheeeep) [reply README.md:158]: Modified.

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Pull request merged**